### PR TITLE
Put common aliases in Map/Set instead of switch over strings

### PR DESCRIPTION
**Pull request by sebmarkbage**
This is a follow up to https://github.com/facebook/react/pull/26546

This is strictly a perf optimization since we know that switches over strings aren't optimally implemented in current engines. Basically they're a sequence of ifs.

As a result, we're better off putting the unusual cases in a Map and the very common cases in the beginning of the switch. We might be better off putting very common cases in explicit ifs - just in case the engine does optimize switches to a hash table which is potentially worse.

### CODE REVIEW

File: packages/react-dom-bindings/src/client/ReactDOMComponent.js [hunk 1]
```diff
@@ -2244,828 +2001,132 @@ function diffHydratedGenericElement(
...
-          extraAttributes,
-        );
-        continue;
-      case 'fillOpacity':
+      case 'xHeight':
         hydrateAttribute(
           domElement,
           propKey,
-          'fill-opacity',
+          'x-height',
```
Comment (sophiebits) [inline packages/react-dom-bindings/src/client/ReactDOMComponent.js:2086]: did you mean to have x-height here?

File: packages/react-dom-bindings/src/shared/getAttributeAlias.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+const aliases = new Map([
+  ['acceptCharset', 'accept-charset'],
+  ['htmlFor', 'for'],
+  ['httpEquiv', 'http-equiv'],
+  // HTML and SVG attributes, but the SVG attribute is sensitive.],
```
Comment (sophiebits) [inline packages/react-dom-bindings/src/shared/getAttributeAlias.js:14]:
```suggestion
  // HTML and SVG attributes, but the SVG attribute is case sensitive.],
```

**Review by sophiebits — APPROVED** [review]

File: packages/react-dom-bindings/src/client/ReactDOMComponent.js [hunk 3]
```diff
@@ -1018,6 +794,17 @@ export function setInitialProperties(
...
 
   switch (tag) {
+    case 'div':
+    case 'span':
+    case 'svg':
+    case 'path':
+    case 'a':
+    case 'g':
+    case 'p':
+    case 'li': {
```
Comment (sophiebits) [inline packages/react-dom-bindings/src/client/ReactDOMComponent.js:804]: https://almanac.httparchive.org/en/2021/markup#top-elements
looks like `i` and `ul` are also popular… wonder if it's worth including them? and moving the img/link special case up? and maybe adding meta? (does float go through this path?)

Comment (sebmarkbage) [reply packages/react-dom-bindings/src/client/ReactDOMComponent.js:804]: I guess they're less popular. Didn't show up in my testing. It's interesting that groupings like `ul` and `tbody` or even `tr` inherently are less common than the n items below them.

Float doesn't go through this path.

I tried to keep things somewhat grouped together in groups that make sense. Could maybe group all void elements but I think we're going to end up special casing `img` soon for suspensey images so then we can move it up.

File: packages/react-dom-bindings/src/client/ReactDOMComponent.js (hunk 1 above)
Comment (sophiebits) [reply packages/react-dom-bindings/src/client/ReactDOMComponent.js:2086]: hmm? in ReactDOMComponent I see xHeight as the only call to hydrateAttribute() other than className tabIndex and the namespaced ones

**Pull request merged**